File: examples/ex2/java/Main.java
import com.adacore.libex2.example.ExamplePackage;
import com.adacore.libex2.example.Counter;

import com.adacore.gnatpolyglot.runtime.ada2java.IntegerArray;

public class Main {
    public static void main(String[] args) {
        Counter.Array arr = new Counter.Array(1, 4);
        for (var c : arr)
            c.increment();
        for (var c : arr)
            System.out.print(c.getCount() + ", ");
        System.out.println();

        ExamplePackage.increment(arr);
        for (var c : arr)
            System.out.print(c.getCount() + ", ");
        System.out.println();

        IntegerArray intArr = ExamplePackage.toIntArr(arr);
        for (var i : intArr)
            System.out.print(i + ", ");
        System.out.println();
    }
}

File: examples/ex2/rust/src/main.rs
use ex2::example;
use ex2::ada::arrays::PolyglotArray;

fn main() {
    let mut arr = PolyglotArray::<example::Counter>::new(1, 4);
    for mut e in &arr {
        e.increment();
    }
    for i in arr.begin()..=arr.end() {
        print!("{}, ", arr.get(i).unwrap().get_count());
    }
    println!();

    example::increment(&mut arr);
    for e in &arr {
        print!("{}, ", e.get_count());
    }
    println!();

    let int_arr = example::to_int_arr(&mut arr);
    for e in &int_arr {
        print!("{}, ", e);
    }
    println!();
}

